### PR TITLE
make bcache tests run

### PR DESCRIPTION
**Pull request by zenglg**
Signed-off-by: Linggang Zeng <linggang.zeng@easystack.cn>

### CODE REVIEW

File: lib/testrunner [hunk 1]
```diff
@@ -2,6 +2,8 @@
 #
 # Test wrapper run inside the VM
 
+set -x
```
Comment (koverstreet) [inline lib/testrunner:5]: take this out before committing

File: lib/libktest.sh [hunk 2]
```diff
@@ -383,7 +383,15 @@ start_vm()
...
-    set|grep -vE '^[A-Z]' > "$ktest_tmp/env"
+    set > "$ktest_tmp/env_tmp"
+    readonly_variables="$(readonly | cut -d= -f1 | cut -d' ' -f3)"
+    for variable in ${readonly_variables}
+    do
+        grep -v "${variable}" "$ktest_tmp/env_tmp" > "$ktest_tmp/env"
+        cp "$ktest_tmp/env" "$ktest_tmp/env_tmp"
+    done
+    sed -i "s/^ ;$//g" "$ktest_tmp/env"
+    rm -rf "$ktest_tmp/env_tmp"
```
Comment (koverstreet) [inline lib/libktest.sh:394]: neat

File: lib/testrunner [hunk 3]
```diff
@@ -14,7 +16,9 @@ KERNEL_ARCH=""
 ktest_dir="/host/$ktest_dir"
 ktest_tmp="/host/$ktest_tmp"
 
-ln -s $ktest_dir /ktest
+if [ ! -e "/ktest" ]; then
+    ln -s $ktest_dir /ktest
+fi
```
Comment (koverstreet) [inline lib/testrunner:21]: why make this a conditional? we're always running in a brand new vm here

**Review by koverstreet — COMMENTED** [review]
main thing is the set -x thing, that'll clutter up the output too much

and I don't like extra unneeded conditionals - why'd you add those?

but overall looks good

File: lib/testrunner (hunk 1 above)
Comment (zenglg) [reply lib/testrunner:5]: ok

File: lib/testrunner (hunk 3 above)
Comment (zenglg) [reply lib/testrunner:21]: There are some reboot tests(ex: backingdev_reboot_fio.ktest, backingdev_reboot_fs.ktest, ...), and the directory of ktest have been linked. So do link again would return error and break the test after next reboot.

**Comment by zenglg** [comment]
> main thing is the set -x thing, that'll clutter up the output too much
> 
> and I don't like extra unneeded conditionals - why'd you add those?
> 
> but overall looks good

Just want more output logs to know how the tests run, now I think it's OK for me without set -x.
And thanks for your comments.

**Review by koverstreet — APPROVED** [review]

**Pull request merged**